### ISSUE TITLE
Implement .error and .warning

### ISSUE THREAD

**Issue by boeckmann** [boeckmann/asm6502#21]
make .abort accept the same parameters as .echo but terminate with error code. Also display file and line number info.

**Comment by boeckmann** [comment]
Implemented by 4642ec2.

**Comment by waltje** [comment]
I have both .error (doing the same as your .abort) as well as .warn, the latter of which does not bail out.

Last night I did the first assembly of my system ROM (BASIC + BIOS) on my (ARM) tablet :P








On Monday, April 24, 2023 at 06:33:19 AM EDT, Bernd Böckmann ***@***.***> wrote: 







make .abort accept the same parameters as .echo but terminate with error code. Also display file and line number info.

**Comment by boeckmann** [comment]
Congrats 👍 

Is it running Windows On Arm?

**Comment by waltje** [comment]
Yep.  I generally want all my software to build and run on Windows and Linux, on { x86,x64,arm,arm64 },
although that sometimes takes work.






On Monday, April 24, 2023 at 11:10:27 AM EDT, Bernd Böckmann ***@***.***> wrote: 







Congrats 👍

Is it running Windows On Arm?

**Comment by boeckmann** [comment]
> Yep.  I generally want all my software to build and run on Windows and Linux, on { x86,x64,arm,arm64 }, although that sometimes takes work. 

Great! And I want mine to run on legacy systems. This will complement each other well. :-)

I adjusted mine to support .warning and .error, to stay somewhat compatible.

**Comment by waltje** [comment]
> legacy systems
Well..

My (four) machines are (two) i7-based Dell's and (two) i9-based Dell's.  Two run Win7-x64, two run Win10-x64.  But: I pretty much ONLY run VMware Workstation on those machines - all "work" is done on VMs that either are WinXPSP3, Win7-x64, Win10-x64, Win11-x64 or one of the several Linux ones (usually Mint.)

Added to that, "within" those VMs I can run Qemu-based ARM and ARM64 machines for WoA testing, being Win8.1-arm or Win10-arm64 based.

I just had fun running my 6502 emulator (which runs my own monitor plus EhBasic etc), atop a Win8.1-arm system running in Qemu8, which ran in a Win7-x64 in a VMware VM, running atop an i7 Win7-x64 machine.

Phew! :P

--f







On Monday, April 24, 2023 at 11:52:21 AM EDT, Bernd Böckmann ***@***.***> wrote: 







>  
> Yep. I generally want all my software to build and run on Windows and Linux, on { x86,x64,arm,arm64 }, although that sometimes takes work.
> 

Great! And I want mine to run on legacy systems. This will complement each other well. :-)

I adjusted mine to support .warning and .error, to stay somewhat compatible.
